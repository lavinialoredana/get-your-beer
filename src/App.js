import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1> Get in touch to order a punk beer </h1>
      </header>
    </div>
  );
}

export default App;
