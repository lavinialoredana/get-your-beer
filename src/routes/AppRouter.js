import { createBrowserRouter } from "react-router-dom";
import App from "../pages/AppPage";
import ErrorPage from "../pages/ErrorPage";

const AppRouter = createBrowserRouter([
  {
    path: "/",
    element: <App />,
    errorPageElement: <ErrorPage />,
  },
]);

export default AppRouter;
