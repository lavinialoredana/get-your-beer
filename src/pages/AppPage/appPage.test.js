import { render, screen, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {
  BrowserRouter,
  createMemoryRouter,
  RouterProvider,
} from "react-router-dom";
import App from ".";
import { routesConfig } from "../../routes/AppRouter";

describe("App Page", () => {
  afterEach(() => {
    cleanup();
  });
  // More details hre: https://testing-library.com/
  it("should contain header", () => {
    //Arrange
    render(<App />, { wrapper: BrowserRouter });
    // Act
    const appHeaderElement = screen.getByTestId("app-header");
    // Assert
    expect(appHeaderElement).toBeInTheDocument;
  });
  it("should contain link to orders", () => {
    render(<App />, { wrapper: BrowserRouter });

    const ordersLinkElement = screen.getByTestId("orders-link");

    expect(ordersLinkElement).toBeInTheDocument;
  });
  it("should go to orders page when clicking on orders link", async () => {
    const router = createMemoryRouter(routesConfig, {
      initialEntries: ["/"],
    });
    render(
      <RouterProvider router={router}>
        <App />
      </RouterProvider>
    );
    const ordersLinkElement = screen.getByTestId("orders-link");

    expect(ordersLinkElement).toBeInTheDocument;

    await userEvent.click(ordersLinkElement);

    expect(ordersLinkElement).not.toBeInTheDocument;
  });
});
