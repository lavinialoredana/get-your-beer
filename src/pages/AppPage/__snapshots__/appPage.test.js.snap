// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Page to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1
        data-testid="app-header"
      >
         Get in touch to order a punk beer 
      </h1>
    </header>
    <a
      data-testid="orders-link"
    >
      GO to Orders
    </a>
    <div
      class="description-container"
    >
      <textarea
        cols="4"
        data-testid="description"
        label="description"
        rows="50"
      />
    </div>
  </div>
</DocumentFragment>
`;
